src/file.cr

```diff
@@ -376,12 +376,16 @@ class File < IO::FileDescriptor
   # File.write("bar", "foo")
   # File.read("bar") # => "foo"
   # ```
+  # Note: this method can't be used to read files bigger than `Int32::MAX` (2GB)
   def self.read(filename, encoding = nil, invalid = nil) : String
     File.open(filename, "r") do |file|
       if encoding
         file.set_encoding(encoding, invalid: invalid)
         file.gets_to_end
       else
+        if file.size > Int32::MAX
```

Comment:
Shouldn't this check be performed *always* and not just when no encoding is defined?

I believe it will be more consistent behavior that `File.read` checks size on any scenario, providing or not an encoding for the file being read.